add_me_to_your_PATH/Dockerfile.template

```diff
@@ -76,13 +52,80 @@ RUN curl -sSL https://raw.githubusercontent.com/beacon-biosignals/github-token-h
     chmod +x $HOME/.github-token-helper && \
     git config --global credential.https://github.com.helper "$HOME/.github-token-helper -f /run/secrets/github_token"
 
+# Switch the Julia depot to use the shared cache storage. As `.ji` files reference
+# absolute paths to their included source files care needs to be taken to ensure the depot
+# path used during package precompilation matches the final depot path used in the image.
+# If a source file no longer resides at the expected location the `.ji` is deemed stale and
+# will be recreated.
+RUN ln -s /tmp/julia-cache ~/.julia
+
 # Install the General registry and optionally a private registry
 ARG PRIVATE_REGISTRY_URL=""
-RUN --mount=type=secret,id=github_token \
+RUN --mount=type=cache,sharing=locked,target=/tmp/julia-cache \
+    --mount=type=secret,id=github_token \
     julia -e 'using Pkg; \
               !isempty(ENV["PRIVATE_REGISTRY_URL"]) && Pkg.Registry.add(RegistrySpec(url=ENV["PRIVATE_REGISTRY_URL"])); \
               Pkg.Registry.add("General")'
 
+#####
+##### `deps` stage
+#####
+
+# The sysimage stage is designed to be invalidated as infrequently as
+# possible while making a sysimage for faster Julia load times.
```

Comment:
does this comment still apply to this stage or the next one?

Reply:
This should be moved

Reply:
Actually, should be removed. It was copied from the `sysimage-image` stage